modules/core/src/main/java/org/apache/ignite/internal/util/IgniteUtils.java

```diff
@@ -12854,4 +12854,20 @@ public static Socket acceptServerSocket(ServerSocket srvrSock) throws IOExceptio
             }
         }
     }
+
+    public static boolean isLambda(Class<?> objectClass) {
```

Comment:
Let's create some unit tests for this method (at least that it can distinguish lambda a lambda from a non-lambda class)